### DISCUSSION TITLE
production ready?

### DISCUSSION THREAD

**Discussion by ghost** [DioxusLabs/dioxus#1117] [General]
Would you run this library in production today? If not, what are your current hesitations?

**Comment by ealmloff** [comment]
I think production readiness depends a lot on what kind of application you are targeting. If you could provide more details about the platforms you are targeting and the kind of application you are making I could give a more specific answer. If you are building a traditional website with largely text/html based content, you should be able to use Dioxus without many issues.

# Stability

Dioxus is not stable. We try to avoid needless breaking changes, but there will be breaking changes before 1.0. There are also likely more bugs than you would encounter in larger frameworks like react because Dioxus has a smaller userbase to find bugs.

# Platform Limitations

## Liveview

Liveview is missing reconnect handlers for when the web socket disconnects, and a router integration

## Web/Desktop

The web and desktop renderers are more mature, but there are still areas we can improve. Mainly building cross platform APIs (https://github.com/DioxusLabs/dioxus-std) and supporting element specific extensions for things like video playback.

## Mobile

Mobile currently does not compile on IOS because of rfd (see #1121)

## SSR

Dioxus currently does not have an equivalent to suspense for server side rendering which can make working with futures difficult for purely server side rendered applications. If hydration is enabled, all futures will resolve on the client

# Other projects

There are a few large projects using Dioxus in production:
1) https://github.com/Satellite-im/Uplink
2) https://github.com/terhechte/Ebou

**Comment by ghost** [comment]
I should hold off for a while before diving deeper into this possibility. I will close this comment for now since I am not actively pursuing this question.
